### ISSUE TITLE
Infinite loop RAML 0.8

### ISSUE THREAD

**Issue by menduz** [raml-org/raml-js-parser-2#235]
[Connectivity_Service-v0.1 (1).zip](https://github.com/raml-org/raml-js-parser-2/files/278365/Connectivity_Service-v0.1.1.zip)

It's on RAML 0.8 and it freezes chrome, safari and firefox.

**Comment by ddenisenko** [comment]
Can not reproduce with browser version of the parser 0.2.20.

The parsing passes.

**Comment by ddenisenko** [comment]
Could you provide more details of how exactly you call the parser?

**Comment by menduz** [comment]
I don't know the version of the parser because it's the one deployed on anypoint api-manager. You can upload the zip of the issue to api-manager and it will fail once you open the editor.

**Comment by ddenisenko** [comment]
Debugged API Designer, found and fixed the bug in 0.2.21 version of the parser.

Note 1: It looks like to make resources to appear in the right panel of API designer, parser should report no errors. To fix ConnectivityService so the parser does not report any errors, the traits description in the bottom of the file should fixed.

Replace
```RAML
traits:
  - secured:
      headers:
        Authorization:
          type: string
          description: |
            Used to send a valid Core Services Access Token
            Authorization: Bearer ACCESS_TOKEN
          required: true
  paged:
        queryParameters:
        ...
```

with 

```RAML
traits:
  - secured:
      headers:
        Authorization:
          type: string
          description: |
            Used to send a valid Core Services Access Token
            Authorization: Bearer ACCESS_TOKEN
          required: true
  - paged:
        queryParameters:
        ...
```

(add dash in the beginning of `paged`).

Note 2: Parser and API Designer are two different products. I do not know when the new version of the parser will be integrated into API Designer.

Note 3: It is hard to debug the packaged version of the parser. I also had to install the designer and all the dependencies locally. It would be good in future to get bugs reproduction steps based directly on the parser, not on dependent products.

**Comment by menduz** [comment]
Thanks
